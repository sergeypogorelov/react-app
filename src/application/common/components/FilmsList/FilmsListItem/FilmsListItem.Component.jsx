import React from 'react';
import { Link } from 'react-router-dom';

import './FilmsListItem.scss';

export default class FilmsListItem extends React.Component {

    render() {
        var i = this.props.item;

        return (
            <div className="film">
                <div className="film-poster">
                    <Link to={ '/film/' + i.name }>
                        <img className="film-poster-image" src={ i.image } />
                    </Link>
                </div>
                <div className="film-title clearfix">
                    <h4 className="film-title-content">{ i.title }</h4>
                    <p className="film-title-year">{ i.year }</p>
                </div>
                <p className="film-genre">{ i.genre }</p>
            </div>
        );
    }

}