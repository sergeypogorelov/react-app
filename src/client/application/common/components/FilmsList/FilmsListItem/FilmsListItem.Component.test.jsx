import React from 'react';
import { HashRouter as Router } from 'react-router-dom';

import renderer from 'react-test-renderer';

import FilmsListItem from './FilmsListItem.Component';

test('Component FilmsListItem works fine.', () => {

    const props = {
        item: {
            name: 'name',
            image: 'image',
            title: 'title',
            year: 'year',
            genre: 'genre'
        }
    };

    const component = renderer.create(
        <Router>
            <FilmsListItem { ...props } />
        </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});