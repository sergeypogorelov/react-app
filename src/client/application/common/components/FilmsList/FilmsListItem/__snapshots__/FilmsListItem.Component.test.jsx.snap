// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FilmsListItem works fine. 1`] = `
<div
  className="film"
>
  <div
    className="film-poster"
  >
    <a
      href="#/film/name"
      onClick={[Function]}
    >
      <img
        className="film-poster-image"
        src="image"
      />
    </a>
  </div>
  <div
    className="film-title clearfix"
  >
    <h4
      className="film-title-content"
    >
      title
    </h4>
    <p
      className="film-title-year"
    >
      year
    </p>
  </div>
  <p
    className="film-genre"
  >
    genre
  </p>
</div>
`;
